### PR TITLE
feat: add hook scripts to pod

### PR DESCRIPTION
**Pull request by wrenix**
# Pull Request

## Description of the change

first step or prepare to #466 

thanks for this hint @jessebot 
https://github.com/nextcloud/helm/issues/466#issuecomment-1857616753

## Benefits

not needed to create a own secret and complex mount per extraVolumes needed to setup such hooks like in:
https://github.com/nextcloud/docker#auto-configuration-via-hook-folders

descripted 

## Checklist 

- [x] DCO has been [signed off on the commit](https://docs.github.com/en/github/authenticating-to-github/signing-commits).
- [x] Chart version bumped in `Chart.yaml` according to [semver](http://semver.org/).
- [ ] (optional) Variables are documented in the README.md

### CODE REVIEW

File: charts/nextcloud/values.yaml [hunk 1]
```diff
@@ -146,7 +146,15 @@ nextcloud:
   #          )
   #        )
   #      );
-
+  # Hooks for auto configuration
+  # Here you could write small scripts which are placed in `/docker-entrypoint-hooks.d/<hook-name>/<hook-name>.sh`
```
Comment (provokateurin) [inline charts/nextcloud/values.yaml:150]:
```suggestion
  # Here you could write small scripts which are placed in `/docker-entrypoint-hooks.d/<hook-name>/helm.sh`
```
They are always mounted like this, no?

File: charts/nextcloud/templates/secrets-hooks.yaml [hunk 2]
```diff
@@ -0,0 +1,19 @@
+{{- if not (values .Values.nextcloud.hooks | compact | empty) }}
+---
+apiVersion: v1
+kind: Secret
```
Comment (provokateurin) [inline charts/nextcloud/templates/secrets-hooks.yaml:4]: Can you explain why you chose Secret over a simple ConfigMap?

File: charts/nextcloud/values.yaml (hunk 1 above)
Comment (wrenix) [reply charts/nextcloud/values.yaml:150]: Correct (i change it after writing docu)

File: charts/nextcloud/templates/secrets-hooks.yaml (hunk 2 above)
Comment (wrenix) [reply charts/nextcloud/templates/secrets-hooks.yaml:4]: I like to write an script/hook to setup oidc (which contains client-secret).

or other tokens and secrets could be in the scripts

Comment (wrenix) [reply charts/nextcloud/templates/secrets-hooks.yaml:4]: (but yes i could change it to ConfigMap and use therefore environment variables)

Comment (provokateurin) [reply charts/nextcloud/templates/secrets-hooks.yaml:4]: I see. I think this should just be a ConfigMap and then you can use the environment variables for the real secrets.

**Review by provokateurin — APPROVED** [review]

**Comment by provokateurin** [comment]
Just need to solve the conflicts

**Pull request merged**